### PR TITLE
Update documentation with ipycmc and organizing scenarios

### PR DESCRIPTION
**Pull request by abarciauskas-bgse**

### CODE REVIEW

**Comment by abarciauskas-bgse** [comment]
I think gitlab is down

>curl: (28) Failed to connect to mas.maap-project.org port 443: Connection timed out

Updates on this forthcoming, but there was a security vulnerability found.

File: docs/source/user_data/create-datasets-for-dashboard.ipynb [hunk 1]
```diff
@@ -8,18 +8,31 @@
     "\n",
     "The following notebook steps through how to add a dataset to the MAAP Dashboard.\n",
     "\n",
-    "The MAAP dashboard has 3 environments:\n",
+    "Note, there are 2 scenarios:\n",
+    "\n",
+    "1. Adding a single Cloud-Optimized GeoTIFF (COG) and,\n",
+    "2. Adding multiple spatially distinct COGs as a \"mosaic\" with mosaicJSON.\n",
```
Comment (wildintellect) [inline docs/source/user_data/create-datasets-for-dashboard.ipynb:14]: they don't actually need to be spatially distinct, overlap is handled via a parameter. But point taken that it results in a single mosaic not a multi-dimensional (e.g. time series).

File: docs/source/user_data/create-datasets-for-dashboard.ipynb [hunk 2]
```diff
@@ -8,18 +8,31 @@
...
+    "1. Inspect your COG(s) to understand the best rescale and colormap name parameters. Optionally create a mosaic.\n",
+    "2. Define a colormap. Colormaps provide mappings of data values to RGB values.\n",
+    "3. Create a PR to the datasets repo to add or update your dataset.\n",
+    "\n",
+    "The MAAP dashboard has 3 environments:\n",
     "\n",
-    "Pre-requisites:\n",
+    "1. Developer-in-test (DIT): biomass.dit.maap-project.org\n",
+    "2. Staging: biomass.staging.maap-project.org / uat.earthdata.nasa.gov/maap-biomass\n",
+    "3. Production: biomass.staging.maap-project.org / earthdata.nasa.gov/maap-project\n",
```
Comment (wildintellect) [inline docs/source/user_data/create-datasets-for-dashboard.ipynb:26]: Should we clarify which url is the public facing/share url?

Comment (philvarner) [reply docs/source/user_data/create-datasets-for-dashboard.ipynb:26]:
```suggestion
    "3. Production: biomass.maap-project.org / earthdata.nasa.gov/maap-biomass\n",
```

File: docs/source/user_data/create-datasets-for-dashboard.ipynb [hunk 3]
```diff
@@ -8,18 +8,31 @@
...
     "\n",
-    "* jupyter notebook\n",
-    "* rasterio, rio-cogeo, supermercado"
+    "These instructions will guide you towards adding your dataset to `bimoass.dit.maap-project.org`. The MAAP Dashboard team will \"promote\" changes to staging and production periodically (release schedule forthcoming)."
```
Comment (philvarner) [inline docs/source/user_data/create-datasets-for-dashboard.ipynb:28]:
```suggestion
    "These instructions will guide you towards adding your dataset to `biomass.dit.maap-project.org`. The MAAP Dashboard team will \"promote\" changes to staging and production periodically (release schedule forthcoming)."
```

File: docs/source/user_data/create-datasets-for-dashboard.ipynb [hunk 4]
```diff
@@ -140,15 +261,72 @@
...
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "# Test the link with the titiler"
+    "## Using mosaicJSON with titiler\n",
+    "\n",
+    "There are 2 options for using mosaicJSON with titiler:\n",
+    "\n",
+    "1. (Preferred) Post mosaicJSON to titiler `mosaics` endpoint and use the `mosaicjson/mosaics` endpoint for dynamic tiling.\n",
+    "2. Upload mosaicJSON to S3 and pass the S3 url to the titiler `tiles` endpoint."
```
Comment (wildintellect) [inline docs/source/user_data/create-datasets-for-dashboard.ipynb:322]: I find `tiles` confusing in this context, `mosaicjson/tiles` might be more clear

**Review by wildintellect — DISMISSED** [review]
Aside from the error @philvarner caught in the url this PR looks good and improves the document. I think we should do a follow up with some better explanations in some parts. 

* As written many people may try to use the same git branch to submit.
* The addition of x at the front of url f string and then replacement seems unnecessary
* Since we don't use WMTS in the dashboard should we switch to using XYZ in the internal checks? Is ipycmc working well enough?
* More screenshots, or visuals previewing colors.
* recursive file scanning should be in Scenario 2, not setup, also should have option to save results for future reuse (DRY)
* Explain when and why to verify cogs
* MosaicJSON from 1 to 16 might actually perform poorly depending on number of files in mosaic, as the json file becomes extremely large.

**Review by wildintellect — CHANGES_REQUESTED** [review]
fix the production urls before merging

**Comment by abarciauskas-bgse** [comment]
@wildintellect addressed comments - can you add your additional feedback to a new card in trello?

**Review by wildintellect — APPROVED** [review]

**Pull request merged**